src/runtime/pprof/pprof.go

```diff
@@ -451,7 +453,55 @@ func printCountCycleProfile(w io.Writer, countName, cycleName string, records []
 
 // printCountProfile prints a countProfile at the specified debug level.
 // The profile will be in compressed proto format unless debug is nonzero.
+// Non-zero debug values have the following behaviors:
+//   - debug=1: text format, aggregated by unique stack, state and labels.
+//   - debug=2: handled elsewhere; the output of runtime.Stack().
+//   - debug=3: binary proto format-identical to debug=0 with per-goroutine labels.
 func printCountProfile(w io.Writer, debug int, name string, p countProfile) error {
+	// Debug >1 is non-aggregated, so branch immediately to avoid computing the
+	// aggregated map if we do not need it. This does duplicate a little of the
+	// logic for actually constructing and outputting the proto profile but only
+	// the helper calls so it is minimal and avoids a more complex structuring of
+	// one output path that handles both the aggregated and non-aggregated cases.
+	if debug == 3 {
+		// Output profile in protobuf form.
+		b := newProfileBuilder(w)
+		b.pbValueType(tagProfile_PeriodType, name, "count")
+		b.pb.int64Opt(tagProfile_Period, 1)
+		b.pbValueType(tagProfile_SampleType, name, "count")
+
+		values := []int64{1}
+		var locs []uint64
+		for i := 0; i < p.Len(); i++ {
+			locs = b.appendLocsForStack(locs[:0], p.Stack(i))
+			var g profilerecord.StackRecord
+			if goroutines, ok := p.(*runtimeProfile); ok {
+				g = goroutines.stk[i]
+			}
+			extra := func() {
+				// If the stackrecord has goroutine details (starting with ID), use them
+				// to synthesize labels.
+				if g.ID != 0 {
+					b.pbLabelNum(tagSample_Label, "go::goroutine", int64(g.ID))
+					if g.ID != 1 {
+						b.pbLabelNum(tagSample_Label, "go::goroutine_created_by", int64(g.CreatorID))
+					}
+					b.pbLabel(tagSample_Label, "go::goroutine_state", pprof_gStatusString(g.State, g.WaitReason), 0)
+					if mins := pprof_gWaitFor(g.State, g.WaitSince); mins > 0 {
```

Comment:
ooo, we can get finer granularity than minutes, right? Are we tempted? I'm kind of tempted. Often I am interested in the "seconds" kind of wait durations. Maybe show the seconds if in `[1,59]` and round to minutes after that?